draft-ietf-tls-esni.md

```diff
@@ -930,6 +959,28 @@ in-band when possible, such as through the use of OCSP stapling, and clients
 SHOULD take steps to minimize or protect such requests during certificate
 validation.
 
+## Abuse of ECH Acceptance Signal
+
+To signal acceptance, the backend server overwrites 8 bytes of its
+ServerHello.random with a value derived from the ClientHelloInner.random. (See
+{{backend-server-behavior}} for details.) In the presence of an active
+adversary, this behavior increases the likelihood of the ServerHello.random
+colliding with the ServerHello.random of a previous session, potentially
+reducing the overall security of the protocol. However, the remaining 24 bytes
+provide enough entropy to ensure this is not a practical avenue of attack.
+
+On the other hand, the probability that two 8-byte strings are the same is
+non-negligible. This poses a modest operational risk. Suppose the client-facing
+server terminates the connection (i.e., ECH is rejected or bypassed): if the
+last 8 bytes of its ServerHello.random coincide with the confirmation signal,
+then the client will incorrectly presume acceptance and proceed as if the
+backend server terminated the connection.
```

Comment:
Can we be explicit with the collision probability? (RFC8446 reserves two sentinel values for the downgrade signals.) We might also note that this probability is equivalent to that of network or transport failures, so in practice it's not much of a concern.

Reply:
> Can we be explicit with the collision probability?

Done.

> (RFC8446 reserves two sentinel values for the downgrade signals.)

Not sure what this comment is about.

> We might also note that this probability is equivalent to that of network or transport failures, so in practice it's not much of a concern.

Would you mind suggesting text for this? I'm not sure I know enough about TCP to be accurate about this.

Reply:
Ah, missed this. I'll file an issue to add it!